torch/distributed/distributed_c10d.py

```diff
@@ -288,9 +288,11 @@ def __init__(self, backend: Backend):
         if backend == Backend.UNDEFINED:
             # default config when backend is not specified
             # supported since PyTorch 2.0
-            for device in Backend.default_device_backend_map:
-                if is_backend_available(Backend.default_device_backend_map[device]):
-                    self.device_backend_map[device] = Backend(Backend.default_device_backend_map[device])
+            for device, default_backend in Backend.default_device_backend_map.items():
```

Comment:
are we making an assumption on which order the map stores the devices in? (i didn't check if its an ordered dict (where the order could still potentially be edited) or if its a dealers choice map)

Reply:
How is the order relevant here? `default_device_backend_map` is a dict str->backend and `device_backend_map` is too.

So I don't see any assumptions or even relevance of the order and even if there was it isn't changed by this PR

Reply:
sorry, you're right. It looks like this would work.